python/paddle/distributed/fleet/base/role_maker.py

```diff
@@ -869,7 +870,8 @@ def _generate_role(self):
             else:
                 self._collective_env()
             self._role_is_generated = True
-            self._gloo_init()
+            if not paddle.fluid.framework.in_dygraph_mode():
```

Comment:
it seems in dygraph mode, we do not need to do gloo_init, but actually it is initialized in other places. In general, this is not a good design, and should be refactored with init_parallel_env in the future.